ext/six.py

```diff
@@ -844,10 +844,71 @@ def wrapper(cls):
                 orig_vars.pop(slots_var)
         orig_vars.pop('__dict__', None)
         orig_vars.pop('__weakref__', None)
+        if hasattr(cls, '__qualname__'):
+            orig_vars['__qualname__'] = cls.__qualname__
         return metaclass(cls.__name__, cls.__bases__, orig_vars)
     return wrapper
 
 
+def ensure_binary(s, encoding='utf-8', errors='strict'):
+    """Coerce **s** to six.binary_type.
+
+    For Python 2:
+      - `unicode` -> encoded to `str`
+      - `str` -> `str`
+
+    For Python 3:
+      - `str` -> encoded to `bytes`
+      - `bytes` -> `bytes`
+    """
+    if isinstance(s, text_type):
+        return s.encode(encoding, errors)
+    elif isinstance(s, binary_type):
+        return s
+    else:
+        raise TypeError("not expecting type '%s'" % type(s))
+
+
+def ensure_str(s, encoding='utf-8', errors='strict'):
+    """Coerce *s* to `str`.
+
+    For Python 2:
+      - `unicode` -> encoded to `str`
+      - `str` -> `str`
+
+    For Python 3:
+      - `str` -> `str`
+      - `bytes` -> decoded to `str`
+    """
+    if not isinstance(s, (text_type, binary_type)):
+        raise TypeError("not expecting type '%s'" % type(s))
+    if PY2 and isinstance(s, text_type):
+        s = s.encode(encoding, errors)
+    elif PY3 and isinstance(s, binary_type):
+        s = s.decode(encoding, errors)
+    return s
+
+
+def ensure_text(s, encoding='utf-8', errors='strict'):
```

Comment:
@medariox we could use thuis more often
?

Instead copying the code, Just call the function?

Reply:
I tried to use it, see: https://github.com/pymedusa/Medusa/pull/5904/commits/46f3fb0ec54e0a1d9288869428228cc3a31e38cc#diff-aa9b79281f0255fad57b25fd6ef5675aR111

It breaks the dredd tests: https://travis-ci.org/pymedusa/Medusa/jobs/471468760#L518-L533

Reply:
Just so this discussion, and if you want to fix this all you have to do is change `.travis.yml`:
https://github.com/pymedusa/Medusa/blob/60b72413447c57f675ad842c357ee4e230257307/.travis.yml#L62
and
https://github.com/pymedusa/Medusa/blob/60b72413447c57f675ad842c357ee4e230257307/.travis.yml#L77

Both have to be changed to `pip install --upgrade six`.
The tests failed because Travis pre-installs `pip` on the containers:
```
$ pip install six
Requirement already satisfied: six in /home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages (1.11.0)
```